spec/requests/user_actions_controller_spec.rb

```diff
@@ -99,5 +99,40 @@
       end
     end
 
+    context "other users' activity" do
+      let(:another_user) { Fabricate(:user) }
+
+      UserAction.private_types.each do |action_type|
+        action_name = UserAction.types.key(action_type)
+        it "users cannot list other users' actions of type: #{action_name}" do
+          list_and_check(action_type, 404)
+        end
+      end
+
+      UserAction.private_types.each do |action_type|
+        action_name = UserAction.types.key(action_type)
+        it "moderators cannot list other users' actions of type: #{action_name}" do
+          sign_in(Fabricate(:moderator))
+          list_and_check(action_type, 404)
+        end
+      end
+
+      UserAction.private_types.each do |action_type|
+        action_name = UserAction.types.key(action_type)
+        it "admins can list other users' actions of type: #{action_name}" do
+          sign_in(Fabricate(:admin))
```

Comment:
Minor but we cant rely on `fab!` to fabricate the `admin` user once and reuse across the tests.